### PR TITLE
Fix bug in check_estimates_on_outputs

### PR DESCRIPTION
**Pull request by psuriana**

### CODE REVIEW

File: src/AutoSchedule.cpp [hunk 1]
```diff
@@ -112,18 +112,25 @@ struct FStage {
...
+            bool found = false;
+            for (int i = (int)estimates.size() - 1; i >= 0; --i) {
+                if ((estimates[i].var == arg) && estimates[i].min.defined() &&
+                    estimates[i].extent.defined()) {
+                    found = true;
+                    est = estimates[i];
+                    break;
+                }
             }
+            user_assert(found && est.min.as<IntImm>() && est.extent.as<IntImm>())
```
Comment (steven-johnson) [inline src/AutoSchedule.cpp]: Does it have to be an IntImm, or merely have type().is_int()?

File: src/AutoSchedule.cpp [hunk 2]
```diff
@@ -679,14 +686,14 @@ map<string, Box> get_pipeline_bounds(DependenceAnalysis &analysis,
...
-                if (est.var == arg) {
+                if ((est.var == arg) && est.min.defined() && est.extent.defined()) {
                     Interval I = Interval(est.min, simplify(est.min + est.extent - 1));
                     pure_bounds.emplace(arg, I);
                     out_box.push_back(I);
                     break;
                 }
             }
-            internal_assert(i >= 0);
+            internal_assert(i >= 0) << "Could not find estimate for " << arg << "\n";
```
Comment (steven-johnson) [inline src/AutoSchedule.cpp]: Should this be user_assert?

**Comment by steven-johnson** [comment]
Can we add a test?

File: src/AutoSchedule.cpp (hunk 2 above)
Comment (psuriana) [reply src/AutoSchedule.cpp]: If check_estimates_on_outputs is working as expected, the assertion should never trigger within the get_pipeline_bounds

**Pull request merged**